### DISCUSSION TITLE
实现文章不在首页里面中显示

### DISCUSSION THREAD

**Discussion by gaohaotian93** [volantis-x/hexo-theme-volantis#421] [Q&A] [answered]
## 自检报告

- [ x] 我已阅读主题文档，确认不存在此功能。
- [ x ] 我已阅读Hexo官方文档，确认不存在此功能。

## 需求描述
让文章不在主页里面中显示
比如hexo n title的文章
在title的front-matter中设置一个是否显示在主页的选项，可以自定义显示
有的文章不希望在主页中显示，可以在归档，标签，分类中找到

**Comment by stevenjoezhang** [comment, marked as answer]
可以安装插件： https://github.com/next-theme/hexo-generator-indexed
这个功能放在主题中实现会有潜在的问题。

**Comment by xaoxuu** [comment]
我觉得 layout 的时候加个判断就可以了

**Comment by stevenjoezhang** [comment]
修改 layout 有一个小问题是，如果某一页隐藏了文章，后一页的文章是不会填补这个空缺的，导致每一页上的文章数目不同。直接对 Hexo 的 generator 下手会比较方便。

**Comment by xaoxuu** [comment]
👍🏼 厉害

**Comment by xaoxuu** [comment]
我的置顶确实也存在数目不同的问题

**Comment by stevenjoezhang** [comment]
文章置顶也是一个大坑 😂

<img width="986" alt="截屏2020-08-08 下午4 26 34" src="https://user-images.githubusercontent.com/16272760/89705929-eeeab300-d993-11ea-8e3e-51771e6cc64c.png">

`hexo-generator-index` 下个版本（2.0.0）会内置这个功能： https://github.com/hexojs/hexo-generator-index/pull/51
由主题实现这个功能确实挺麻烦的

[a comment by an automated account is not shown]
